components/kyma-environment-broker/internal/ias/client.go

```diff
@@ -0,0 +1,235 @@
+package ias
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	kebError "github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/error"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+)
+
+const (
+	PathServiceProviders  = "/service/sps"
+	PathCompanyGlobal     = "/service/company/global"
+	PathAccess            = "/service/sps/%s/rba"
+	PathIdentityProviders = "/service/idp"
+	PathDelete            = "/service/sps/delete"
+)
+
+type (
+	ClientConfig struct {
+		URL    string
+		ID     string
+		Secret string
+	}
+
+	Client struct {
+		config         ClientConfig
+		httpClient     *http.Client
+		closeBodyError error
+	}
+
+	Request struct {
+		Method  string
+		Path    string
+		Body    io.Reader
+		Headers map[string]string
+	}
+)
+
+func NewClient(cli *http.Client, cfg ClientConfig) *Client {
+	return &Client{
+		config:     cfg,
+		httpClient: cli,
+	}
+}
+
+func (c *Client) SetType(spID string, payload Type) error {
+	return c.call(c.serviceProviderPath(spID), payload)
+}
+
+func (c *Client) SetAssertionAttribute(spID string, payload PostAssertionAttributes) error {
+	return c.call(c.serviceProviderPath(spID), payload)
+}
+
+func (c *Client) SetSubjectNameIdentifier(spID string, payload SubjectNameIdentifier) error {
+	return c.call(c.serviceProviderPath(spID), payload)
+}
+
+func (c *Client) SetAuthenticationAndAccess(spID string, payload AuthenticationAndAccess) error {
+	pathAccess := fmt.Sprintf(PathAccess, spID)
+
+	return c.call(pathAccess, payload)
+}
+
+func (c *Client) GetCompany() (*Company, error) {
```

Comment:
```suggestion
func (c *Client) GetCompany() (_ *Company,  err error) {
```

the one important thing is to declare error in signature so the defer statement will work properly. 

Same for other places